src/main/java/ProjectOcean/Controller/YearController.java

```diff
@@ -109,6 +151,66 @@ private int calculateSlot(double y) {
         }
     }
 
+    @Override
+    public void update(Observable o, Object arg) {
+       // System.out.println("UPDATE");
+        updateCoursesInMap();
+        displayAllCoursesInStudyPlan();
+    }
+
+    private void updateCoursesInMap(){
```

Comment:
This method could be cleaned up from comments/empty lines.

Reply:
Indeed, will fix that.